dubbo-rpc/dubbo-rpc-api/src/main/java/com/alibaba/dubbo/rpc/RpcResult.java

```diff
@@ -95,10 +95,8 @@ public boolean hasException() {
         return attachments;
     }
 
-    public void setAttachments(Map<String, String> map) {
-        if (map != null && map.size() > 0) {
-            attachments.putAll(map);
-        }
+    public void setAttachments(Map<String, String> attachments) {
+        this.attachments = attachments == null ? new HashMap<String, String>() : attachments;
```

Comment:
I dont suggest this modified code,

Reply:
getAttachments() can get this map, and removeAll it to achieve this replacement and these modified codes:
1、change an empty map when param is null
2、modify the meanings of  setAttachments

Reply:
setAttachments() is not referenced by other codes currently, so the cost of the change is relatively low. Should we change `setAttachments()` and add `addAttachements()` so that we can keep consistent with that in RpcInvocation?

Reply:
I think the original impl has no problem, it is just a convenient API to add attachment in batch. I will close this PR and add the following javadoc upon this method:

```java
    /**
     * Append all items from the map into the attachment, if map is empty then nothing happens
     *
     * @param map contains all key-value pairs to append
     */
    public void setAttachments(Map<String, String> map) {
        if (map != null && map.size() > 0) {
            attachments.putAll(map);
        }
    }
```